ui/appui-react/src/appui-react/widget-panels/Content.tsx

```diff
@@ -37,21 +37,36 @@ function WidgetFallback() {
 export function WidgetContent() {
   const widget = useWidgetDef();
   const itemId = widget?.id ?? widget?.label ?? "unknown";
+  const prevDocumentRef = React.useRef<Document>(document);
   const onSave = React.useCallback(() => {
     widget?.saveTransientState();
   }, [widget]);
   const onRestore = React.useCallback(() => {
     widget?.restoreTransientState();
+
+    const el = ref.current;
+    if (!el) return;
+    const prevDocument = prevDocumentRef.current;
+    if (prevDocument === el.ownerDocument) return;
+
+    const ev = new CustomEvent("appui:reparent", { bubbles: true });
```

Comment:
Are there other cases where reparenting happens? e.g. maybe moving to a different node within the same document? If yes, then it would be best to fire this event unconditionally whenever reparenting happens.

Reply:
Good point, I was too focused on our use-case of reparenting between documents.
Yes, when docking/undocking the widget.

Reply:
Removed the `ownerDocument` comparison check.